test/integration/operator/clustermanager_aws_test.go

```diff
@@ -77,5 +80,27 @@ var _ = ginkgo.Describe("ClusterManager Default Mode with aws registration", fun
 				return annotation == "arn:aws:iam::123456789012:role/hub-cluster_managed-cluster-identity-creator"
 			}, eventuallyTimeout, eventuallyInterval).Should(gomega.BeTrue())
 		})
+		ginkgo.It("should have auto approved arn patterns separated by comma with awsirsa", func() {
+			gomega.Eventually(func() bool {
+				registrationControllerDeployment, err := kubeClient.AppsV1().Deployments(hubNamespace).
+					Get(context.Background(), fmt.Sprintf("%s-registration-controller", clusterManagerName), metav1.GetOptions{})
+				if err != nil {
+					return false
+				}
+				commandLineArgs := registrationControllerDeployment.Spec.Template.Spec.Containers[0].Args
+				autoApprovedArnPatterns, present := findMatchingArg(commandLineArgs, "--auto-approved-arn-patterns")
+				return present && strings.SplitN(autoApprovedArnPatterns, "=", 2)[1] ==
+					"arn:aws:eks:us-west-2:123456789013:cluster/.*,arn:aws:eks:us-west-2:123456789012:cluster/.*"
+			}, eventuallyTimeout, eventuallyInterval).Should(gomega.BeTrue())
+		})
 	})
 })
+
+func findMatchingArg(args []string, pattern string) (string, bool) {
+	for _, commandLineArg := range args {
+		if strings.Contains(commandLineArg, pattern) {
```

Comment:
nit: probably should do the exact equals check here in case you have multiple arguments with similar strings. ie: `--foo-bar-foo`  `--foo-bar` and doing a `--foo-bar` will just grab the first one which is `--foo-bar-foo`.